src/v2/cookbook/form-validation.md

```diff
@@ -0,0 +1,322 @@
+  </p>
+
+  <p>
+    Current Total: {{total}}
+  </p>
+
+  <p>
+  <input type="submit" value="Submit">  
+  </p>
+
+</form>
+```
+
+Note the set of inputs covering the five different features. Note the addition of .number to the v-model attibute. This tells Vue to cast the value to a number when you use it. However, there is a bug with this feature such that when the value is blank, it turns back into a string. You'll see the workaround below. To make it a bit easier for the user, we also added a current total right below so they can see, in real time, what their total is. Now let's look at the JavaScript.
+
+```
+const app = new Vue({
+  el:'#app',
+  data:{
+    errors:[],
+    weapons:0,
+    shields:0,
+    coffee:0,
+    ac:0,
+    mousedroids:0
+  },
+  computed:{
+     total:function() {
+       //must parse cuz Vue turns empty value to string
+       return Number(this.weapons)+
+         Number(this.shields)+
+         Number(this.coffee)+
+         Number(this.ac+this.mousedroids);
+     }
+  },
+  methods:{
+    checkForm:function(e) {
+      this.errors = [];
+      if(this.total != 100) this.errors.push("Total must be 100!");
+      if(!this.errors.length) return true;
+      e.preventDefault();
+    }
+  }
+})
+```
+
+We set up the total value as a computed value, and outside of that bug I ran into, it was simple enough to setup. My checkForm method now just needs to see if the total is 100 and that's it. You can play with this here:
+
+<p data-height="265" data-theme-id="0" data-slug-hash="vWqGoy" data-default-tab="html,result" data-user="cfjedimaster" data-embed-version="2" data-pen-title="form validation 3" class="codepen">See the Pen <a href="https://codepen.io/cfjedimaster/pen/vWqGoy/">form validation 3</a> by Raymond Camden (<a href="https://codepen.io/cfjedimaster">@cfjedimaster</a>) on <a href="https://codepen.io">CodePen</a>.</p>
+<script async src="https://production-assets.codepen.io/assets/embed/ei.js"></script>
+
+## Server-side Validation
+
+In my final example, we built something that makes use of Ajax to validate at the server. The form will ask you to name a new product and will then check to ensure that the name is unique. We wrote a quick OpenWhisk serverless action to do the validation. While it isn't terribly important, here is the logic:
+
+``` js
+function main(args) {
+
+    return new Promise((resolve, reject) => {
+
+        // bad product names: vista,empire,mbp
+        let badNames = ['vista','empire','mbp'];
+        if(badNames.includes(args.name)) reject({error:'Existing product'});
+        resolve({status:'ok'});
+
+    });
+
+}
+```
+
+Basically any name but "vista", "empire", and "mbp" are acceptable. Ok, so let's look at the form.
+
+``` html
+<form id="app" @submit="checkForm" method="post">
+  
+  <p v-if="errors.length">
+    <b>Please correct the following error(s):</b>
+    <ul>
+      <li v-for="error in errors">{{ error }}</li>
+    </ul>
+  </p>
+
+  <p>

```

Comment:
Again, it's a small thing but please format the HTML in tandem with the DOM :) thanks